includes/Abstracts/OrderMigration.php

```diff
@@ -0,0 +1,540 @@
+     */
+    public function clear_dokan_vendor_balance_table( $order_id ) {
+        global $wpdb;
+        $wpdb->delete(
+            $wpdb->prefix . 'dokan_vendor_balance', array(
+                'trn_id'   => $order_id,
+                'trn_type' => 'dokan_orders',
+            )
+        );
+    }
+
+    /**
+     * Removes existing data from dokan refund table.
+     *
+     * @since 1.0.0
+     *
+     * @param int $order_id
+     * @param int $seller_id
+     *
+     * @return void
+     */
+    public function clear_dokan_refund_table( $order_id ) {
+        global $wpdb;
+        $wpdb->delete( $wpdb->prefix . 'dokan_refund', array( 'order_id' => $order_id ) );
+    }
+
+    /**
+     * Updates dokan order table with new order id.
+     *
+     * @since 1.0.0
+     *
+     * @param array $dokan_order_data
+     * @param int $sub_order_id
+     * @param int $seller_id
+     * @param Object $order_obj
+     *
+     * @return void
+     */
+    public function sync_dokan_order_table( $dokan_order_data, $sub_order_id, $seller_id, $order_obj ) {
+        $order_total = $order_obj->get_total();
+        $net_amount = $dokan_order_data['net_sale'];
+        $created_date = reset( $dokan_order_data['commission_data'] )['created'];
+
+        global $wpdb;
+
+        $wpdb->insert(
+            $wpdb->prefix . 'dokan_orders',
+            array(
+                'order_id'     => $sub_order_id,
+                'seller_id'    => $seller_id,
+                'order_total'  => $order_total,
+                'net_amount'   => $net_amount,
+                'order_status' => 'wc-' . $order_obj->get_status(),
+            ),
+            array(
+                '%d',
+                '%d',
+                '%f',
+                '%f',
+                '%s',
+            )
+        );
+
+        $wpdb->insert(
+            $wpdb->prefix . 'dokan_vendor_balance',
+            array(
+                'vendor_id'     => $seller_id,
+                'trn_id'        => $sub_order_id,
+                'trn_type'      => 'dokan_orders',
+                'perticulars'   => 'New order',
+                'debit'         => $net_amount,
+                'credit'        => 0,
+                'status'        => 'wc-' . $order_obj->get_status(),
+                'trn_date'      => $created_date,
+                'balance_date'  => gmdate( 'Y-m-d h:i:s', strtotime( $created_date ) ),
+            ),
+            array(
+                '%d',
+                '%d',
+                '%s',
+                '%s',
+                '%f',
+                '%f',
+                '%s',
+                '%s',
+                '%s',
+            )
+        );
+    }
+
+    /**
+     * Creates sub orders for each seller.
+     *
+     * @since 1.0.0
+     *
+     * @param int $seller_id
+     * @param array $seller_products
+     *
+     * @return \WC_Order
+     */
+    public function create_sub_order( $seller_id, $seller_products ) {
+        $bill_ship = array(
```

Comment:
This method is redundant. The default `dokan()->order->create_sub_order()` could be used to do this.
Follow the below steps and also check if the given steps satisfy the needs.
```
    /**
     * Creates sub orders for each seller.
     *
     * @since 1.0.0
     *
     * @param int $seller_id
     * @param array $seller_products
     *
     * @return \WC_Order
     */
    public function create_sub_order( $seller_id, $seller_products ) {
        /*
         * Before creating sub order, we need to convert the meta key according to
         * Dokan meta key for order items as those data will be parsed while creating
         * sub orders.
         */
        $this->map_shipping_method_item_meta();

        return dokan()->order->create_sub_order( $this->order, $seller_id, $seller_products );
    }

    /**
     * Converts the order item meta key according to
     * Dokan meta key as those data will be parsed
     * while creating sub orders.
     *
     * @since 1.0.0
     *
     * @return void
     */
    public function map_shipping_method_item_meta() {
        if ( ! $this->order instanceof \WC_Order ) {
            return;
        }

        $shipping_methods = $this->order->get_shipping_methods();
        if ( empty( $shipping_methods ) ) {
            return;
        }

        foreach ( $shipping_methods as $method_item_id => $shipping_object ) {
            wc_update_order_item_meta(
                $method_item_id,
                'seller_id',
                wc_get_order_item_meta( $method_item_id, 'vendor_id', true )
            );
        }
    }
```
And if it works, the `create_shipping` method inside this class will not be needed.